### PR TITLE
fix: add timeout to retry wipe IPMI commands (PXE + powercycle)

### LINKED ISSUE #222: Sidero: retry IPMI operations

### PR DESCRIPTION
**Pull request by smira**
As server is about to be powercycled via IPMI, we set a condition on a
server to prevent duplicate attempts to powercycle while the wipe is
going, but this condition was never retried.

In order to workaround IPMI bugs, we now retry powercycling after
`WipeTimeout`. Also reconcile servers in !Clean state to make sure we
have a chance to retry.

Other small changes:

* bump Talos to 0.7.0
* add another check to make sure cluster is fully provisioned before
calling it ready (sfyra)

Fixes #222

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by Unix4ever — DISMISSED** [review]

File: app/metal-controller-manager/controllers/server_controller.go [hunk 1]
```diff
@@ -170,13 +171,14 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 				r.Recorder.Event(serverRef, corev1.EventTypeNormal, "Server Management", "Server powered on and set to PXE boot once.")
 			}
 
+			conditions.Delete(&s, metalv1alpha1.ConditionPowerCycle)
```
Comment (smira) [inline app/metal-controller-manager/controllers/server_controller.go:174]: add comment?

File: app/metal-controller-manager/controllers/server_controller.go [hunk 2]
```diff
@@ -114,35 +113,37 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
 	case s.Status.InUse && !s.Status.IsClean:
-		return f(true, false)
+		return f(true, ctrl.Result{})
 	case !s.Status.InUse && !s.Status.IsClean:
-		if conditions.Has(&s, metalv1alpha1.ConditionPowerCycle) && conditions.IsFalse(&s, metalv1alpha1.ConditionPowerCycle) {
-			return f(false, false)
+		if conditions.Has(&s, metalv1alpha1.ConditionPowerCycle) &&
+			conditions.IsFalse(&s, metalv1alpha1.ConditionPowerCycle) &&
+			time.Since(conditions.GetLastTransitionTime(&s, metalv1alpha1.ConditionPowerCycle).Time) < constants.WipeTimeout {
+			return f(false, ctrl.Result{RequeueAfter: constants.WipeTimeout / 5})
```
Comment (smira) [inline app/metal-controller-manager/controllers/server_controller.go:131]: add comment

File: app/metal-controller-manager/pkg/constants/constants.go [hunk 3]
```diff
@@ -14,4 +14,6 @@ const (
 	InitrdAsset = "initramfs.xz"
 
 	DefaultRequeueAfter = time.Second * 20
+
+	WipeTimeout = time.Hour
```
Comment (smira) [inline app/metal-controller-manager/pkg/constants/constants.go:18]: report from the agent every X seconds that wipe is working

Comment (smira) [reply app/metal-controller-manager/pkg/constants/constants.go:18]: created note/issue for it

**Review by andrewrynhard — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**